IdentityCore/src/validation/MSIDAuthorityFactory.m

```diff
@@ -0,0 +1,96 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAuthorityFactory.h"
+#import "MSIDAADAuthority.h"
+#import "MSIDADFSAuthority.h"
+#import "MSIDB2CAuthority.h"
+
+@implementation MSIDAuthorityFactory
+
+- (MSIDAuthority *)authorityFromUrl:(NSURL *)url
+                               context:(id<MSIDRequestContext>)context
+                                 error:(NSError **)error
+{
+    if ([MSIDB2CAuthority isAuthorityFormatValid:url context:context error:nil])
+    {
+        __auto_type b2cAuthority = [[MSIDB2CAuthority alloc] initWithURL:url context:context error:nil];
+        if (b2cAuthority) return b2cAuthority;
+    }
+    
+    if ([MSIDADFSAuthority isAuthorityFormatValid:url context:context error:nil])
+    {
+        __auto_type adfsAuthority = [[MSIDADFSAuthority alloc] initWithURL:url context:context error:nil];
+        if (adfsAuthority) return adfsAuthority;
+    }
+    
+    if ([MSIDAADAuthority isAuthorityFormatValid:url context:context error:nil])
+    {
+        __auto_type aadAuthority = [[MSIDAADAuthority alloc] initWithURL:url context:context error:nil];
+        if (aadAuthority) return aadAuthority;
+    }
+    
+    if (error)
+    {
+        *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"Provided authority url is not a valid authority.", nil, nil, nil, context.correlationId, nil);
+        
+        MSID_LOG_ERROR(context, @"Provided authority url is not a valid authority.");
+    }
+    
+    return nil;
+}
+
+- (MSIDAuthority *)authorityFromUrl:(NSURL *)url
+                          rawTenant:(NSString *)rawTenant
+                            context:(id<MSIDRequestContext>)context
+                              error:(NSError **)error
+{
+    if ([MSIDB2CAuthority isAuthorityFormatValid:url context:context error:nil])
+    {
+        __auto_type b2cAuthority = [[MSIDB2CAuthority alloc] initWithURL:url rawTenant:rawTenant context:context error:nil];
+        if (b2cAuthority) return b2cAuthority;
+    }
+    
+    if ([MSIDAADAuthority isAuthorityFormatValid:url context:context error:nil])
```

Comment:
this checks for MSIDAADAuthority and creates ADFS authority

Reply:
fixed